controllers/host/host_controller.go

```diff
@@ -1877,14 +1877,17 @@ func (r *HostReconciler) ReconcileResource(
 		if utils.ContainsString(instance.ObjectMeta.Finalizers, HostFinalizerName) {
 			// A finalizer is still present so we need to try to delete the
 			// host from the system.
-			if host != nil {
+			if host == nil {
+				logHost.Info("host being deleted is no longer present on system")
+			} else {
 				err = r.ReconcileDeletedHost(client, instance, host)
 				if err != nil {
+					instance.ObjectMeta.Finalizers = utils.RemoveString(instance.ObjectMeta.Finalizers, HostFinalizerName)
```

Comment:
Those tasks are identical to line 1894-1897, why to execute them here in error case why they will be eventually triggered later?

Reply:
If ReconcileDeletedHost method is returning error then the finalizer is not removed through the code
Which is resulting in the stuck of the kubectl delete host 
So before returning if the finalizer is removed the host gets deleted

Reply:
In this case, don't we need to remove the finalizer before deleting the host? Why throw an error out?

Reply:
we need to delete the finalizer  to execute the kubectl delete host command successfully
SO the code is updated accordingly